### PR TITLE
Add CSV backend

### PR DESCRIPTION
**Pull request by tsai00**
Added support for local CSV DB (included in XlsxDB)

### CODE REVIEW

File: dbhydra/examples/transactions_and_build_query_example.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+
+from dbhydra.src.tables import MysqlTable
+
+
+table1=MysqlTable(db1,"prices") #database needs to be specified
+table1.query_building_enabled=True
+table1.select("price")
```
Comment (Omastto1) [inline dbhydra/examples/transactions_and_build_query_example.py:7]: whats the point of this line?

File: dbhydra/src/abstract_table.py [hunk 2]
```diff
@@ -237,11 +239,16 @@ def __init__(self, db1, name, columns = None, types: Optional[list[str]] = None,
     # Temporary disabled, please make sure this is implemented where needed, don't introduce breaking changes please
     # @abc.abstractmethod
     @classmethod
-    def init_from_column_type_dict(cls, db1, name, column_type_dict, id_column_name="id"):
+    def init_from_column_type_dict(cls, db1, name, column_type_dict, id_column_name="id", use_csv=False):
```
Comment (DovaX) [inline dbhydra/src/abstract_table.py:242]: could we avoid this?

Comment (DovaX) [reply dbhydra/src/abstract_table.py:242]: I would like the signature of the function to be the same for all dialects

File: dbhydra/src/abstract_table.py [hunk 3]
```diff
@@ -237,11 +239,16 @@ def __init__(self, db1, name, columns = None, types: Optional[list[str]] = None,
...
         column_converted_type_dict = db1._convert_column_type_dict_from_python(column_type_dict)
         columns = list(column_converted_type_dict.keys())
         types = list(column_converted_type_dict.values())
-        return cls(db1, name, columns, types, id_column_name=id_column_name)
+
+        # FIXME: not ideal from OOP prospective
+        if cls == XlsxTable:
```
Comment (DovaX) [inline dbhydra/src/abstract_table.py:248]: ahhhh, can we do this differently?

File: dbhydra/src/tables.py [hunk 4]
```diff
@@ -880,28 +878,30 @@ def modify_column(self, column_name, new_column_type):
...
     def drop(self):
-        if self.table_directory_path.exists():
-            if sys.platform in ["darwin","linux","linux2"]:
-                command="rm "+str(self.table_directory_path)
-            else:
-                command="del "+str(self.table_directory_path)
-            os.system(command)
-        
-
+        self.table_directory_path.unlink(missing_ok=True)
```
Comment (DovaX) [inline dbhydra/src/tables.py:904]: how does this work?

Comment (DovaX) [reply dbhydra/src/tables.py:904]: should have comment

Comment (DovaX) [reply dbhydra/src/tables.py:904]: Are you sure its equivalent to drop the folder structure?

**Review by DovaX — CHANGES_REQUESTED** [review]

File: dbhydra/src/tables.py (hunk 4 above)
Comment (tsai00) [reply dbhydra/src/tables.py:904]: This is not function for removing a folder, but rather for removing a file (see `self.table_directory_path` definition, the name though is misleading and could be rather `self.table_path`, but I just kept the original one). Anyway `unlink` method will remove file.

Comment (tsai00) [reply dbhydra/src/tables.py:904]: And overall calling os-specific commands directly is not the best practice, especially when there is a dedicated library for that :)

File: dbhydra/src/abstract_table.py (hunk 2 above)
Comment (tsai00) [reply dbhydra/src/abstract_table.py:242]: Didn't find a quick way to do that, tried also to have an overloaded method for XlsxTable but that did not solve the problem

File: dbhydra/src/tables.py (hunk 4 above)
Comment (DovaX) [reply dbhydra/src/tables.py:903]: Ah you are right - it's confusing name - it shouldnt be named "table_directory_path"

File: dbhydra/src/xlsx_db.py [hunk 5]
```diff
@@ -13,8 +13,9 @@ class XlsxDb(AbstractDb):
     
     matching_table_class = XlsxTable
     
-    def __init__(self, config_file="config.ini", db_details=None):
+    def __init__(self, config_file="config.ini", db_details=None, is_csv=False):
```
Comment (DovaX) [inline dbhydra/src/xlsx_db.py:16]: here I am still not fully happy but ok for now

**Review by DovaX — APPROVED** [review]

**Pull request merged**